src/Algebra/Graph/Bipartite/AdjacencyMap.hs

```diff
@@ -678,15 +687,144 @@ vertexSet = Set.fromAscList . vertexList
 edgeSet :: (Ord a, Ord b) => AdjacencyMap a b -> Set.Set (a, b)
 edgeSet = Set.fromAscList . edgeList
 
+-- | The sorted /adjacency list/ of the left part of the graph.
+-- Complexity: /O(n + m)/ time and /O(n)/ memory.
+--
+-- @
+-- leftAdjacencyList 'empty'            == []
+-- leftAdjacencyList ('vertices' [] xs) == []
+-- leftAdjacencyList ('vertices' xs []) == [ (x, []) | x <- nub ('sort' xs) ]
+-- leftAdjacencyList ('edge' x y)       == [(x, [y])]
+-- leftAdjacencyList ('star' x ys)      == [(x, 'nub' ('sort' ys))]
+-- @
+leftAdjacencyList :: AdjacencyMap a b -> [(a, [b])]
+leftAdjacencyList (BAM lr _) = [ (v, Set.toAscList us) | (v, us) <- Map.toAscList lr ]
+
+-- | The sorted /adjacency list/ of the right part of the graph.
+-- Complexity: /O(n + m)/ time and /O(n)/ memory.
+--
+-- @
+-- rightAdjacencyList 'empty'            == []
+-- rightAdjacencyList ('vertices' [] xs) == [ (x, []) | x <- 'nub' ('sort' xs) ]
+-- rightAdjacencyList ('vertices' xs []) == []
+-- rightAdjacencyList ('edge' x y)       == [(y, [x])]
+-- rightAdjacencyList ('star' x ys)      == [ (y, [x])  | y <- 'nub' ('sort' ys) ]
+-- @
+rightAdjacencyList :: AdjacencyMap a b -> [(b, [a])]
+rightAdjacencyList (BAM _ rl) = [ (v, Set.toAscList us) | (v, us) <- Map.toAscList rl ]
+
+-- | The sorted /adjacency list/ of the graph.
+-- Complexity: /O(n + m)/ time and /O(n)/ memory.
+--
+-- @
+-- adjacencyList 'empty'            == []
+-- adjacencyList ('vertices' xs ys) == [ (v, []) | v <- (map Left (sort (nub xs))) ++ (map Right (sort (nub ys))) ]
+-- adjacencyList ('edge' x y)       == [(Left x, [Right y]), (Right y, [Left x])]
+-- adjacencyList                  == 'Algebra.Graph.AdjacencyMap.adjacencyList' . 'fromBipartite'
+-- @
+adjacencyList :: AdjacencyMap a b -> [(Either a b, [Either a b])]
```

Comment:
This function looks a bit awkward because `Left`s are only paired with `Right`s and vice versa. The user of this function will likely need to use some unsafe constructions to extract the contents. 

Perhaps, `leftAdjacencyList` and `rightAdjacencyList` would be sufficient for the API.